dtfs-central-api/src/v1/controllers/tfm/deal/tfm-get-deals.controller.js

```diff
@@ -92,102 +61,122 @@ const findDeals = async (searchString, sortBy, fieldQueries, callback) => {
+     * Query/filter deals by any custom field
+     * - I.e, date/timestamp fields, deals created by X bank.
+     * - All single value string fields are supported.
+     * - However, only specific timestamp fields are supported.
+     * - This is as per business requirements. More timestamp fields can be easily added if required.
+     * - Timestamp field queries are currently only used by an external system (Cedar).
+     */
+    fieldQueries.forEach((field) => {
+      const { name: fieldName, value: fieldValue } = field;
+
+      if (isTimestampField(fieldName)) {
+        // NOTE: A deal timestamp field could be at any time of the day.
+        // Given a date from fieldQueries in the format 'dd-MM-yyyy', we:
+        // 1) generate timestamps for the start and end of that day
+        // 2) check that the requested timestamp field falls within this particular day;
+        // ...with the use of $gte and $lte.
+        // Example:
+        // - given: '01-11-2021'
+        // - dayStartTimestamp will be: 1636329600000
+        // - dayEndTimestamp will be: 1636415999999
+        // - if a deal has e.g submissionDate timestamp that is within this day (e.g 1636378182935.0)
+        // the deal will be returned in the MongoDB query.
+
+        const { dayStartTimestamp, dayEndTimestamp } = dayStartAndEndTimestamps(fieldValue);
+
+        query = {
+          ...query,
+          [fieldName]: {
+            $gte: dayStartTimestamp,
+            $lte: dayEndTimestamp,
+          },
+        };
+      } else {
+        query = {
+          ...query,
+          [fieldName]: {
+            $eq: fieldValue,
+          },
+        };
+      }
+    });
   }
-
-  deals = dealsArray;
-
-  if (sortBy) {
-    deals = sortDeals(deals, sortBy);
+  const doc = await dealsCollection
+    .aggregate([
+      {
+        $match: query,
+      },
+      {
+        $addFields: {
+          sortId: {
+            $cond: {
+              if: { $eq: ['$dealSnapshot.dealType', CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS] },
+              then: sortBy ? `$${sortBy.bssField}` : undefined,
+              else: sortBy ? `$${sortBy.field}` : undefined,
+            }
+          }
+        }
+      },
+      {
+        $sort: {
+          ...selectedField,
+          updatedAt: -1,
+          _id: 1,
+        },
+      },
+      {
+        $unset: 'sortId'
+      },
+      {
+        $facet: {
+          count: [{ $count: 'total' }],
+          deals: [{ $skip: page * (pagesize ?? 0) }, ...(pagesize ? [{ $limit: parseInt(pagesize, 10) }] : [])],
+        },
+      },
+      { $unwind: '$count' },
+      {
+        $project: {
+          count: '$count.total',
+          deals: true,
+        },
+      },
+    ])
+    .toArray();
+
+  if (!doc.length) {
+    const pagination = {
+      totalItems: 0,
+      currentPage: page,
+      totalPages: 0,
+    };
+    return { deals: [], pagination };
   }
+  const { count, deals } = doc[0];
 
-  if (callback) {
-    callback(deals);
-  }
```

Comment:
Note to self: double check that we don't need the option to pass in a callback anymore